pystapi-client/src/pystapi_client/client.py

```diff
@@ -0,0 +1,283 @@
+import re
+import urllib
+import urllib.parse
+import warnings
+from collections.abc import Callable, Iterable, Iterator
+from typing import (
+    Any,
+    cast,
+)
+
+import httpx
+from pydantic import AnyUrl
+from stapi_pydantic import Link, Product
+from stapi_pydantic.product import ProductsCollection
+
+from pystapi_client.conformance import ConformanceClasses
+from pystapi_client.exceptions import APIError
+from pystapi_client.stapi_api_io import StapiIO
+from pystapi_client.warnings import NoConformsTo
+
+DEFAULT_LINKS = [
+    {
+        "endpoint": "/conformance",
+        "rel": "conformance",
+        "method": "GET",
+    },
+    {
+        "endpoint": "/products",
+        "rel": "products",
+        "method": "GET",
+    },
+]
+
+
+class Client:
+    """A Client for interacting with the root of a STAPI
+
+    Instances of the ``Client`` class provide a convenient way of interacting
+    with STAPI APIs that conform to the `STAPI API spec
+    <https://github.com/stapi-spec/stapi-spec>`_.
+    """
+
+    _stapi_io: StapiIO
+    _conforms_to: list[str] = []
+    _extra_fields: dict[str, Any] = {}
```

Comment:
These are just set in the `__init__`, so don't need to be set at the class level.